src/component/mxgraph/MxGraphRenderer.ts

```diff
@@ -30,34 +30,20 @@ export default class MxGraphRenderer {
   constructor(readonly graph: mxGraph, readonly coordinatesTranslator: CoordinatesTranslator, readonly styleConfigurator: StyleConfigurator) {}
 
   public render(bpmnModel: BpmnModel): void {
+    const displayedModel = toDisplayedModel(bpmnModel);
+
     const model = this.graph.getModel();
     model.clear(); // ensure to remove manual changes or already loaded graphs
     model.beginUpdate();
-
-    const collapsedSubProcessIds: string[] = bpmnModel.flowNodes
-      .filter(shape => {
-        const bpmnElement = shape.bpmnElement;
-        return ShapeUtil.isSubProcess(bpmnElement?.kind) && (bpmnElement as ShapeBpmnSubProcess)?.markers.includes(ShapeBpmnMarkerKind.EXPAND);
-      })
-      .map(shape => shape.bpmnElement?.id);
-
     try {
-      this.insertShapes(bpmnModel.pools);
-      this.insertShapes(bpmnModel.lanes);
-      this.insertShapes(bpmnModel.flowNodes.filter(shape => ShapeUtil.isSubProcess(shape.bpmnElement?.kind)));
-      this.insertShapes(
-        bpmnModel.flowNodes.filter(shape => {
-          const kind = shape.bpmnElement?.kind;
-          return !ShapeUtil.isBoundaryEvent(kind) && !ShapeUtil.isSubProcess(kind) && !collapsedSubProcessIds.includes(shape.bpmnElement?.parentId);
-        }),
-      );
-      this.insertShapes(
-        bpmnModel.flowNodes.filter(shape => {
-          const kind = shape.bpmnElement?.kind;
-          return ShapeUtil.isBoundaryEvent(shape.bpmnElement?.kind) && !ShapeUtil.isSubProcess(kind) && !collapsedSubProcessIds.includes(shape.bpmnElement?.parentId);
-        }),
-      );
-      this.insertEdges(bpmnModel.edges);
+      this.insertShapes(displayedModel.pools);
```

Comment:
👍 It's easier to read, but instead of adding a new class, I will have rather modify the BPMNModel, since we already distinguish between Lane / Pool / Flownode in the shapes.

Reply:
Why not but this means removing from the model children of the collapsed subprocess.
I don't see any impacts for now for the rendering but I haven't think about future features that could be impacted by such a change.

Reply:
Can we do such a refactoring in another PR. With the current changes, everything occur only in the Render file?
Changes directly in the model will have more impacts, so to keep this PR foccused on the fix, this is probably better to not limit changes.

Reply:
No problem 😄